test/types/object.js

```diff
@@ -673,6 +673,38 @@ describe('object', () => {
             });
         });
 
+        it('renames using a regular expression', (done) => {
+
+            const regex = /foobar/i;
+
+            const schema = Joi.object({
+                fooBar: Joi.string()
+            }).rename(regex, 'fooBar');
+
+            Joi.compile(schema).validate({ FOOBAR: 'a' }, (err, value) => {
+
+                expect(err).to.not.exist();
+                done();
+            });
+        });
+
+        it('using regex and override enabled it should allow overwriting existing value', (done) => {
+
+            const regex = /foobar/i;
+
+            const schema = Joi.object({
+                fooBar: Joi.string(),
+                fooBaz: Joi.string()
+            }).rename(regex, 'fooBar', { override: true });
```

Comment:
Once we fix the usage of `alias` we should update this test and also add a test for the `override` functionality.

I would expect that if we have multiple keys matching our regex and override set to `false` then we would throw an error. Similarly if we have `override` set to `false` and the property is already on the object we would throw an error.